src/config_inference.ts

```diff
@@ -0,0 +1,209 @@
+        spinner = wait(
+          `Guessing project name '${projectName}': creating project...`,
+        )
+          .start();
+      } else {
+        spinner = wait("Creating new project with a random name...").start();
+      }
+      try {
+        const project = await api.createProject(projectName);
+        if (projectName) {
+          spinner.succeed(
+            `Guessed project name '${project.name}'.`,
+          );
+        } else {
+          spinner.succeed(`Created new project '${project.name}'`);
+        }
+        wait({ text: "", indent: 3 }).start().info(
+          `You can always change the project name in https://dash.deno.com/projects/${project.name}/settings`,
+        );
+        return project.name;
+      } catch (e) {
+        if (e instanceof APIError && e.code == "projectNameInUse") {
+          spinner.stop();
+          spinner = wait(
+            `Guessing project name '${projectName}': this project name is already used. Checking ownership...`,
+          ).start();
+          const hasAccess = projectName &&
+            (await api.getProject(projectName)) !== null;
+          if (hasAccess) {
+            spinner.stop();
+            const confirmation = confirm(
+              `${
+                magenta("?")
+              } Guessing project name '${projectName}': you already own this project. Should I deploy to it?`,
+            );
+            if (confirmation) {
+              return projectName;
+            }
+          }
+          projectName = `${projectName}-${Math.floor(Math.random() * 100)}`;
+          spinner.stop();
+        } else if (e instanceof APIError && e.code == "slugInvalid") {
+          // Fallback to random name given by the API
+          projectName = undefined;
+          spinner.stop();
+        } else {
+          spinner.fail(
+            `Guessing project name '${projectName}': Creating project...`,
+          );
+          error(e.code);
+        }
+      }
+    }
+  }
+}
+
+async function inferProjectFromOriginUrl() {
+  let originUrl = await getOriginUrlUsingGitCmd();
+  if (!originUrl) {
+    originUrl = await getOriginUrlUsingFS();
+  }
+  if (!originUrl) {
+    return;
+  }
+  const result = originUrl.match(
+    /[:\/]+(?<org>[^\/]+)\/(?<repo>[^\/]+?)(?:\.git)?$/,
+  )?.groups;
+  if (result) {
+    return `${result.org}-${result.repo}`;
+  }
+}
+
+function inferProjectFromCWD() {
+  const projectName = basename(Deno.cwd())
+    .toLowerCase()
+    .replaceAll(/[\s_]/g, "-")
+    .replaceAll(/[^a-z,A-Z,-]/g, "")
+    .slice(0, 26);
+  if (NONAMES.every((n) => n !== projectName)) {
+    return projectName;
+  }
+}
+
+/** Try getting the origin remote URL using the git command */
+async function getOriginUrlUsingGitCmd(): Promise<string | undefined> {
+  const cmd = await new Deno.Command("git", {
+    args: ["remote", "get-url", "origin"],
+  }).output();
+  if (cmd.stdout) {
+    return new TextDecoder().decode(cmd.stdout).trim();
+  }
+}
```

Comment:
If there's no remote named `origin` then the git command just fails with non-zero exit code. In this case stdout is empty. So it'd be better to check the exit code before looking at `cmd.stdout`

Reply:
Empty stdout is falsy

Reply:
To clarify, `stdout` is an empty `UInt8Array` which is not falsy, but it decodes to an empty string which is falsy. Maybe in this case it's overly confusing. I'll be more explicit